src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -638,11 +641,7 @@ public PrimaryKeys apply(PrimaryKeys existing, PrimaryKey neww)
                 return existing;
 
             heapAllocations.add(existing.remove(neww));
-            if (!existing.isEmpty())
-                return existing;
-
-            heapAllocations.add(-PrimaryKeys.unsharedHeapSize());
```

Comment:
should we keep this memory tracking update code?

Reply:
I don't think so, because we return empty PrimaryKeys, so I if I understand correctly, the node will be left in-tree.

Reply:
@pkolaczk is correct. We retain the `PrimaryKeys` object so we don't remove its bytes. Note that on the lines above, we do remove the cost of the reference (however it's possible that this is misleading depending on the way the `PrimaryKeys` object stores references since we might have a fixed cost array somewhere still)